client/my-sites/upgrades/domain-management/dns/dns-list.jsx

```diff
@@ -43,7 +43,11 @@ const DnsList = React.createClass( {
 	addDns: function( record ) {
 		addDnsAction( this.props.selectedDomainName, record, ( error ) => {
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'The DNS record could not be restored.' ) );
+			} else {
+				this.props.successNotice( this.translate( 'The DNS record has been restored.' ), {
+					duration: 5000
```

Comment:
`error.message` is just a string coming from the server (probably from an `WP_Error` that was thrown there). So it should work easily with `errorNotice`, right?

> I had a chat with @mtias about this and we kind of think that it would be best to render <Notice> component somewhere where it is most appropriate visually.

Hmm, I thought that having a common "message bus" was a good idea - since it allowed developers to show the user messages in the same, familiar way, no matter which view they are visiting.
If you just leave us with doing in a way that "is most appropriate visually" we'll get many different implementations of this concept. Or are there any guidelines on the matter? Signaling success/failure of submitting a form seems like one of the most basic and common use cases here :)

PS: no, _you_ go to sleep! I can stay awake as lon-.... zzzZZZzzzzZz :zzz: